### DISCUSSION TITLE
SQL help

### DISCUSSION THREAD

**Discussion by rik1301** [sqlitebrowser/sqlitebrowser#3110] [Q&A]
Hi,
I execute some SQL to remove already processed records from my main table when I do a fresh import.  This worked great as it showed me in the log how many records had been removed.  However, when I added a 2nd table, it only shows how many records were removed from that table, and not the 1st one, even though several thousand records are being removed each time.

Is there a way to show the records removed for both tables, or if not, the total records removed from both tables?

Here is my SQL : 

DELETE FROM table1
WHERE table1.Hex IN (SELECT filtertable.Hex FROM filtertable)
;
DELETE FROM table2
WHERE table2.Hex IN (SELECT filtertable.Hex FROM filtertable)
;

After executing the SQL I get this : 

Execution finished without errors.
Result: query executed successfully. Took 30ms, 2 rows affected
At line 4:
DELETE FROM table2
WHERE table2.Hex IN (SELECT filtertable.Hex FROM filtertable)
;

In this particular case, approx 2000 records were removed from table1, and 2 from table2.  But it only shows me the removal count from table2 and nothing for table1.

?

**Comment by FriedrichFroebel** [comment]
This sounds like a general limitation of DB4S which only shows the query results (and corresponding result counts) for the last query you execute in the "Execute SQL" tab, regardless of whether you have executed multiple statements or just one.

The only way I am aware of to circumvent this is to run each query/line separately (using <kbd>Shift + F5</kbd> instead of <kbd>F5</kbd>), although this might introduce some overhead on your side.
